### PR TITLE
Opsrc 645/added editing controller

### PR DESCRIPTION
**Pull request by milwoz**

### CODE REVIEW

File: src/CommandHandler/Wishlist/UpdateWishlistNameHandler.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        $wishlist = $updateWishlistName->getWishlist();
+
+        $wishlistCookieToken = $this->wishlistCookieTokenResolver->resolve();
+
+        if (null !== $updateWishlistName->getChannelCode()) {
+            $channel = $this->channelRepository->findOneByCode($updateWishlistName->getChannelCode());
+            $wishlist->setChannel($channel);
+        }
+
+        $wishlists = $this->wishlistRepository->findAllByToken($wishlistCookieToken);
```
Comment (christopherhero) [inline src/CommandHandler/Wishlist/UpdateWishlistNameHandler.php:47]: to check if a user doesn't already have such a wishlist (with the same name) you'd better pull it from the database query (i.e. build a query in the repository that takes the name into account)

File: src/Controller/Action/UpdateWishlistNameAction.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        $this->commandBus = $commandBus;
+        $this->flashBag = $flashBag;
+        $this->translator = $translator;
+        $this->channelContext = $channelContext;
+        $this->wishlistRepository = $wishlistRepository;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $wishlistName = $request->request->get('edit_wishlist_name')['name'];
```
Comment (christopherhero) [inline src/Controller/Action/UpdateWishlistNameAction.php:53]: there is a small hole (what if someone doesn't pass on the `name` ? an "Undefined offset: ..." exception will be thrown.

File: src/Controller/Action/UpdateWishlistNameAction.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        $wishlist = $this->wishlistRepository->find($wishlistId);
+
+        try {
+            $channel = $this->channelContext->getChannel();
+        } catch (ChannelNotFoundException $exception) {
+            $channel = null;
+        }
+
+        try {
+            if (null !== $channel) {
```
Comment (christopherhero) [inline src/Controller/Action/UpdateWishlistNameAction.php:64]: this case should not exist

File: src/Controller/Action/UpdateWishlistNameAction.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                $this->translator->trans('bitbag_sylius_wishlist_plugin.ui.wishlist_name_changed')
+            );
+        } catch (HandlerFailedException $exception) {
+            $this->flashBag->add(
+                'error',
+                $this->translator->trans('bitbag_sylius_wishlist_plugin.ui.wishlist_name_already_exists')
+            );
+        }
+
+        return new JsonResponse();
```
Comment (christopherhero) [inline src/Controller/Action/UpdateWishlistNameAction.php:83]: I don't quite understand.
If it's a REST action why do you need a Flashbag ? 
I think if you want create json action you should use api_platform (method) or please change this return statement to wishlist list/grid action

File: src/Resources/assets/shop/js/handleAddAnotherWishlistModal.js [hunk 5]
```diff
@@ -1,18 +1,24 @@
-import { AddWishlistModal } from './addAnotherWishlistModal';
+import { WishlistModal } from './wishlistModal';
```
Comment (christopherhero) [inline src/Resources/assets/shop/js/handleAddAnotherWishlistModal.js:1]: js were checked by another person

**Review by christopherhero — CHANGES_REQUESTED** [review]

File: src/Controller/Action/UpdateWishlistNameAction.php (hunk 3 above)
Comment (milwoz) [reply src/Controller/Action/UpdateWishlistNameAction.php:64]: Removed.

File: src/CommandHandler/Wishlist/UpdateWishlistNameHandler.php (hunk 1 above)
Comment (milwoz) [reply src/CommandHandler/Wishlist/UpdateWishlistNameHandler.php:47]: Done.

File: src/CommandHandler/Wishlist/UpdateWishlistNameHandler.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public function __invoke(UpdateWishlistName $updateWishlistName): void
+    {
+        $wishlist = $updateWishlistName->getWishlist();
+
+        $wishlistCookieToken = $this->wishlistCookieTokenResolver->resolve();
+
+        if (null !== $updateWishlistName->getChannelCode()) {
+            $channel = $this->channelRepository->findOneByCode($updateWishlistName->getChannelCode());
```
[the remark this answers is not in the document]
Comment (milwoz) [reply src/CommandHandler/Wishlist/UpdateWishlistNameHandler.php:43]: Removed.

File: features/editing_name_of_wishlist.feature [hunk 7]
```diff
@@ -8,13 +8,14 @@ Feature: Editing wishlists name
...
-        And I follow edit for "Wishlist1"
-        And I fill in "edit_wishlist_name_name" with "Wishlist2"
-        When I press "edit_wishlist_name_save"
+        When I press "wishlist-edit-button-Wishlist1"
+        And I fill in "edit_wishlist_name" with "Wishlist2"
+        When I press "edit_wishlist_save"
         Then I should be on "/wishlists"
-        And I should see "Wishlist2"
\ No newline at end of file
+        Then I should wait for one second
```
[the remark this answers is not in the document]
Comment (milwoz) [reply features/editing_name_of_wishlist.feature:20]: I done that by reloading site. Would that approach be better?

File: src/Controller/Action/UpdateWishlistNameAction.php (hunk 4 above)
Comment (milwoz) [reply src/Controller/Action/UpdateWishlistNameAction.php]: Done. Returned wishlist list.

File: src/Controller/Action/UpdateWishlistNameAction.php (hunk 2 above)
Comment (milwoz) [reply src/Controller/Action/UpdateWishlistNameAction.php:52]: Done. Added assertion.

**Review by christopherhero — APPROVED** [review]

**Pull request merged**